driver-sync/src/main/com/mongodb/client/internal/MongoClientDelegate.java

```diff
@@ -64,22 +64,21 @@
     private final List<MongoCredential> credentialList;
     private final Object originator;
     private final OperationExecutor operationExecutor;
-    private volatile Crypt crypt;
+    private final Crypt crypt;
 
-    public MongoClientDelegate(final Cluster cluster, final List<MongoCredential> credentialList, final Object originator) {
-        this(cluster, credentialList, originator, null);
+    public MongoClientDelegate(final Cluster cluster, final List<MongoCredential> credentialList, final Object originator,
+                               @Nullable final Crypt crypt) {
+        this(cluster, credentialList, originator, null, crypt);
     }
 
     MongoClientDelegate(final Cluster cluster, final List<MongoCredential> credentialList,
-                        final Object originator, @Nullable final OperationExecutor operationExecutor) {
+                        final Object originator, @Nullable final OperationExecutor operationExecutor,
+                        @Nullable final Crypt crypt) {
         this.cluster = cluster;
         this.serverSessionPool = new ServerSessionPool(cluster);
         this.credentialList = credentialList;
         this.originator = originator;
         this.operationExecutor = operationExecutor == null ? new DelegateOperationExecutor() : operationExecutor;
-    }
-
-    void setCrypt(final Crypt crypt) {
```

Comment:
With the setCrypt method, we could create the delegate, and call setCrypt in MongoClientImpl.init.  But the legacy MongoClient is not an interface so we can't just add a required `init` method to it.  So seemed better to just set the Crypt in the constructor.  But that introduce the problem I highlighted above about not being able to call `getDatabase` until after everything has unwound.